framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1037,7 +1037,7 @@ protected void checkVarargs(AnnotatedExecutableType invokedMethod, Tree tree) {
                 args = ((NewClassTree) tree).getArguments();
                 break;
             default:
-                throw new AssertionError("Unexpected kind of tree: " + tree);
+                throw new AssertionError("Unexpected kind of tree: " + tree); // unreachable
```

Comment:
For this particular kind of error, default cases that should never happen, shouldn't we use `ErrorReporter.errorAbort` so that in a future update we get a better message?